api/app/src/routes/middlewares/oauth2.ts

```diff
@@ -24,6 +24,10 @@ export const processOAuth2 = async (
   const token = await Constants.PROVIDER_OAUTH2_MAP[provider].getTokenFromAuthCode(authCode);
   if (!token) throw new UnauthorizedError();
 
+  // subscribe to webhooks if available
+  const callbackUrl = `${Config.getApiUrl()}/webhook/${provider}`;
+  await Constants.PROVIDER_OAUTH2_MAP[provider].subscribeToWebhooks?.(token, callbackUrl);
+
```

Comment:
To avoid tying webhook logic to OAuth, maybe we can have a "post-auth" function that allows providers to execute an extension to OAuth? Withings' extension is to subscribe to tokens, others might have their own logic, while others don't do anything.

Also, in theory this could be asynchronous and post `updateProviderData()` - maybe even sending the result of that for the extension function?

Reply:
Updated the logic let me know if that works

Reply:
It looks great! The only addition I'd make is to have the call to `postAuth` inside `processOAuth2`, it's not the route's responsibility to know about that logic/flow, and it seems fair to let `processOAuth2` take care of that.